polaris/competition/_competition.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+import os
+from typing import Optional, Union
+
+from pydantic import (
+    field_serializer,
+    field_validator,
+    ValidationInfo
+)
+from polaris.benchmark import BenchmarkSpecification
+from polaris.hub.settings import PolarisHubSettings
+from polaris.utils.types import AccessType, HubOwner, PredictionsType, TimeoutTypes, ZarrConflictResolution
+from polaris.utils.errors import InvalidCompetitionError
+
+
+class CompetitionSpecification(BenchmarkSpecification):
+    """This class extends the [`BenchmarkSpecification`][polaris.benchmark.BenchmarkSpecification] to
+    facilitate interactions with Polaris Competitions.
+
+    Much of the underlying data model and logic is shared across Benchmarks and Competitions, and
+    anything within this class serves as a point of differentiation between the two.
+
+    facilitate interactions with Polaris Competitions.
+
+    Much of the underlying data model and logic is shared across Benchmarks and Competitions, and
+    anything within this class serves as a point of differentiation between the two.
+
+    Currently, these entities will primarily differ at how user predictions are evaluated.
+    """
+
+    # Additional properties specific to Competitions
+    owner: HubOwner
+    start_time: datetime | None = None
+    scheduled_end_time: datetime | None = None
+    actual_end_time: datetime | None = None
+
+    @field_validator("split")
+    def _validate_test_set(cls, split, info: ValidationInfo):
+        """Verifies that the test does not have too many missing values. There must be at
+        least one value per row in the test set across the target columns."""
+        dataset = info.data.get("dataset")
+        target_cols = info.data.get("target_cols")
+        test_indices = split[1]
+
+        if dataset.table.loc[test_indices, target_cols].notna().any(axis=1).all():
+            return split
+        else:
+            raise InvalidCompetitionError("All rows of the test set must have at least one value.")
```

Comment:
We weren't sure if this check was necessary for regular benchmarks, where people might be able to process the test set themselves (e.g. if they wanted to fill in missing values rather than remove them). But it seems reasonable to disallow missing test labels if you think that won't cause too much friction for people trying to upload benchmarks.

Reply:
Ah, I see! To make sure results are comparable, it's important they use the exact same test set. Therefore I would not want to allow e.g. imputing missing test values.